docs/source/ext/adbc_misc.py

```diff
@@ -0,0 +1,238 @@
+        self.env.note_dependency(rel_filename)
+
+        path = Path(filename).resolve()
+        status = driver_status(path)
+
+        generated_lines = [
+            f":bdg-primary:`{status.implementation}`"
+            f":bdg-{status.badge_type}:`{status.status}`"
+        ]
+
+        if status.packages:
+            generated_lines.append("")
+            generated_lines.append("Available from:")
+            for repo, package, url in status.packages:
+                generated_lines.append(
+                    f":bdg-link-secondary:`{repo} ({package}) <{url}>`"
+                )
+
+        if status.implementation in {"C/C++", "C#", "Go", "Rust"}:
+            if not status.packages:
+                generated_lines.append("")
+            generated_lines.append(
+                "May be used from C/C++, C#, GLib, Go, R, "
+                "Ruby, and Rust via the driver manager."
+            )
+
+        parsed = docutils.nodes.Element()
+        nested_parse_with_titles(
+            self.state,
+            StringList(generated_lines, source=""),
+            parsed,
+        )
+        return parsed.children
+
+
+class DriverStatusTableDirective(SphinxDirective):
+    has_content = True
+    required_arguments = 0
+    optional_arguments = 0
+    option_spec: OptionSpec = {}
+
+    def run(self):
+        table = []
+        for line in self.content:
+            if "=>" in line:
+                xref, _, path = line.partition("=>")
+                xref = xref.strip()
+                path = path.strip()
+            else:
+                xref = None
+                path = line.strip()
+
+            if "[#" in path:
+                footnote = path[path.index("[#") + 2 : -1].strip()
+                path = path[: path.index("[#")].strip()
+            else:
+                footnote = None
+
+            rel_filename, filename = self.env.relfn2path(path)
+            self.env.note_dependency(rel_filename)
+
+            path = Path(filename).resolve()
+            status = driver_status(path)
+            table.append((status, xref, footnote))
+
+        table.sort(key=lambda x: (x[0].vendor, x[0].implementation))
+
+        generated_lines = [
+            ".. list-table::",
+            "   :header-rows: 1",
+            "",
+            "   * - Vendor",
+            "     - Implementation",
+            "     - Driver Status",
+            "     - Packages [#packages]_",
+            "",
+        ]
+        for row in table:
+            if row[1]:
+                generated_lines.append(f"   * - :doc:`{row[0].vendor} <{row[1]}>`")
+            else:
+                generated_lines.append(f"   * - {row[0].vendor}")
+
+            if row[2]:
+                generated_lines[-1] += f" [#{row[2]}]_"
+
+            generated_lines.append(f"     - {row[0].implementation}")
+            badge_type = row[0].badge_type
+            generated_lines.append(f"     - :bdg-{badge_type}:`{row[0].status}`")
```

Comment:
```suggestion
            generated_lines.append(f"     - {row[0].status}")
```

The visual weight this has in the table is relatively high. I think badges on the individual driver README pages is fine but I wonder if we can retain the plain-text formatting of the statuses in this table?